File: doc-examples/example-java/src/main/java/example/StudentRepository.java

package example;

import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.Repository;
import io.micronaut.data.annotation.Version;
import io.micronaut.data.repository.CrudRepository;

// tag::studentRepository[]
@Repository
public interface StudentRepository extends CrudRepository<Student, Long> {

    void update(@Id Long id, @Version Long version, String name);

    void delete(@Id Long id, @Version Long version);
}
// end::studentRepository[]

File: doc-examples/mongo-example-python/src/main/python/example/StudentRepository.py
# tag::studentRepository[]
from typing import Annotated

from micronaut.data.annotation import Id, Version
from micronaut.data.mongodb.annotation import MongoRepository
from micronaut.data.repository import CrudRepository
from org.bson.types import ObjectId

from example.Student import Student


@MongoRepository
class StudentRepository(CrudRepository[Student, ObjectId]):

    def update(self, id: Annotated[ObjectId, Id], version: Annotated[int, Version], name: str) -> None: ...

    def delete(self, id: Annotated[ObjectId, Id], version: Annotated[int, Version]) -> None: ...
# end::studentRepository[]
